### ISSUE TITLE
`use-ember-data-rfc-395-imports` should ignore `ember-data/store`

### ISSUE THREAD

**Issue by yads** [ember-cli/eslint-plugin-ember#2156]
According to this deprecation https://deprecations.emberjs.com/id/ember-data-deprecate-legacy-imports `ember-data/store` should still be used instead of `@ember-data/store`. The current rule only ignores `ember-data/types/registries/` (which I'm not even sure is valid anymore). At the very least `ember-data/store` should be added to the ignore list.

**Comment by gorner** [comment]
Agreed, it's confusing as a straightforward re-export like this will not violate the rule:

```es6
export { default } from 'ember-data/store';
```

But any attempts to extend it – as suggested in the [recent EmberConf upgrade talk](https://youtu.be/3_kTmfLG6b0?t=258) and its [accompanying cheat sheet](https://request-service-cheat-sheet.netlify.app) – is considered a violation.

**Comment by Windvis** [comment]
We're also running into this when importing it as a type for our store service declarations.

There are differences between the `ember-data/store` and `@ember-data/store` types (since one is preconfigured) so we do want the "old" import. We're just using disable comments where needed, for now, but a fix for this would be nice.

I made an attempt in #2162.
